sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/_tests_/e2e/application-offering-change-request.aest.controller.assessApplicationOfferingChangeRequest.e2e-spec.ts

```diff
@@ -0,0 +1,198 @@
+  ApplicationOfferingChangeRequestStatus,
+  ApplicationStatus,
+} from "@sims/sims-db";
+import { Repository } from "typeorm";
+
+describe("ApplicationOfferingChangeRequestAESTController(e2e)-assessApplicationOfferingChangeRequest", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+  let applicationOfferingChangeRequestRepo: Repository<ApplicationOfferingChangeRequest>;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    applicationOfferingChangeRequestRepo = dataSource.getRepository(
+      ApplicationOfferingChangeRequest,
+    );
+  });
+
+  it("Should assess the application offering change request for the provided application offering change request id.", async () => {
+    // Arrange
+    const application = await saveFakeApplication(db.dataSource, undefined, {
+      applicationStatus: ApplicationStatus.Completed,
+    });
+    const applicationOfferingChangeRequest =
+      await saveFakeApplicationOfferingRequestChange(
+        db,
+        {
+          application,
+        },
+        {
+          initialValues: {
+            applicationOfferingChangeRequestStatus:
+              ApplicationOfferingChangeRequestStatus.InProgressWithSABC,
+          } as ApplicationOfferingChangeRequest,
+        },
+      );
+    const note = "Some dummy note.";
+    const token = await getAESTToken(AESTGroups.BusinessAdministrators);
+    const endpoint = `/aest/application-offering-change-request/${applicationOfferingChangeRequest.id}`;
+    // Act/Assert
+    await request(app.getHttpServer())
+      .patch(endpoint)
+      .send({
+        note,
+        applicationOfferingChangeRequestStatus:
+          ApplicationOfferingChangeRequestStatus.Approved,
+      })
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK);
+    // Check if the application offering change request was assessed as expected.
+    const updatedApplicationOfferingChangeRequest =
+      await applicationOfferingChangeRequestRepo.findOne({
+        select: {
+          id: true,
+          applicationOfferingChangeRequestStatus: true,
+          assessedNote: { id: true, description: true },
+        },
+        relations: { assessedNote: true },
+        where: { id: applicationOfferingChangeRequest.id },
+      });
+    expect(
+      updatedApplicationOfferingChangeRequest.applicationOfferingChangeRequestStatus,
+    ).toBe(ApplicationOfferingChangeRequestStatus.Approved);
+    expect(
+      updatedApplicationOfferingChangeRequest.assessedNote.description,
+    ).toBe(note);
+  });
+
+  it("Should throw a HttpStatus Not Found (404) error when an application offering change is assessed for an application not in completed state.", async () => {
+    // Arrange
+    const application = await saveFakeApplication(db.dataSource, undefined, {
+      applicationStatus: ApplicationStatus.Overwritten,
```

Comment:
If the intention is validate overwritten applications not being processed, I would say line 104 must have a valid status (InProgressWithSABC) to approve.

Btw. I am not sure if this scenario is required.

Reply:
I have removed this scenario @dheepak-aot. Ideally, this scenario shouldn't arise.